apis/v1alpha1/route_types.go

```diff
@@ -166,5 +168,11 @@ type RouteStatus struct {
 	// manages the Gateway should add an entry to this list when the
 	// controller first sees the route and should update the entry as
 	// appropriate when the route is modified.
+	//
+	// A maximum of 100 Gateways will be represented in this list. If this list
+	// is full, there may be additional Gateways using this Route that are not
+	// included in the list.
+	//
+	// +kubebuilder:validation:MaxItems=100
```

Comment:
Can we stick to powers of 2 here? How about 64 or 128?

Reply:
Is there a strong reason to align to 2^i? Are we sticking these in cache lines or something :^)

Reply:
Although it's an arbitrary limit, I think 100 might be easier for users to understand than 128. 64 does feel a bit too low here.

Reply:
> Is there a strong reason to align to 2^i? Are we sticking these in cache lines or something :^)

There is no reason that I know of. I think it is only about consistency and having limits that users can easily remember and interpret.

Reply:
I think powers of 2 makes a lot of sense for numbers below a certain threshold, maybe that's 100? At least to me, using 128 would feel more out of place/strange than 100 here.

Reply:
I don't feel strongly either ways.